File: WOT-iOS/WOTData/WOTData/core/requests/Session/WOTClearSessionRequest.m
//
//  WOTClearSessionRequest.m
//  WOT-iOS
//
//  Created on 6/5/15.
//  Copyright (c) 2015. All rights reserved.
//

#import "WOTClearSessionRequest.h"
#import <WOTPivot/WOTPivot-Swift.h>
#import <WOTData/WOTData.h>
#import <WOTData/WOTData-Swift.h>

@implementation WOTClearSessionRequest

- (void)temp_executeWithArgs:(WOTRequestArguments *)args {

    [super temp_executeWithArgs:args];
    
    id<WOTCoredataProviderProtocol> dataProvider = [WOTTankCoreDataProvider sharedInstance];
    NSManagedObjectContext *context = [dataProvider mainManagedObjectContext];
    [context performBlock:^{
        
        [UserSession removeObjectsByPredicate:nil inManagedObjectContext:context];

        NSError *error = nil;
        [context save:&error];
        
        if (self.callback) {
            
            self.callback(nil, error, nil);
        }
    }];
}

@end

File: WOT-iOS/WOTData/Sources/webRequests/WOTClearSessionRequest.swift
//
//  WOTClearSessionRequest.swift
//  WOTData
//
//  Created by Pavel Yeshchyk on 3/5/20.
//  Copyright © 2020 Pavel Yeshchyk. All rights reserved.
//

import WOTKit

@objc
public class WOTClearSessionRequest: WOTWEBRequest, WOTModelServiceProtocol {
    override public var method: HTTPMethods { return .POST }

    public static func modelClass() -> PrimaryKeypathProtocol.Type? {
        return nil
    }

    public func instanceModelClass() -> AnyClass? {
        return nil
    }

    @objc
    public class func modelClassName() -> String {
        return ""
    }

    // - (void)start:(WOTRequestArguments * _Nonnull)args {
    //
    //    [super start:args];
    //
    //    id<WOTCoredataProviderProtocol> dataProvider = [WOTTankCoreDataProvider sharedInstance];
    //    NSManagedObjectContext *context = [dataProvider mainManagedObjectContext];
    //    [context performBlock:^{
    //
    //        [UserSession removeObjectsByPredicate:nil inManagedObjectContext:context];
    //
    //        NSError *error = nil;
    //        [context save:&error];
    //
    //        if (self.callback) {
    //
    //            self.callback(nil, error, nil);
    //        }
    //    }];
    // }
}
